### ISSUE TITLE
2019-7-31

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#72]
231.Change to limitations and future work; (Limitation) The prerequisite that we can do optimize to a problem is that there are many solution to this problem. As a result, SC works better on engineering problem than mathematical problems which have only one answer. (2019-6-25)
232.  To see the sentence of limitations and future work; download all papers recommended by Prof. He; let us start our RNN model; Wax, Waa, Wya, ba, by are all in parameters; WTF is softmax; review softmax in the past; use python to implement softmatx; done; go through the dinosaur name; (2019-6-26)
233.Create dinosaur name txt book; rename downloaded papers; done; find read paper method from peng jianming; what is softmax function? Its dimension is equal, also K; logic function is used to describe the S figure; the reason why we use exponent is to make sure that each value is positive; continue on dinosaur; (2019-6-30)
234.There is a difference in function rnn_forward between rnn model and dinosaurs; what is a? as we can see, they name variables using dimensions; replace cache in RNN model; it seems that they use cache to check the result; delete cache in RNN; done; delete parameters; give up; (2019-6-30)
235.Run backprop of RNN; read article about backprop; too hard; review simple backprop; review logic regression; done; modify note of machine learning; done; review neural network; logic regression is classification; logic regression is to choose theta; what is h(theta) in logistic regression; np.exp is exponent; cost function only contain last column, namely output; (2019-7-1)
236.X, Y = create_placeholders(n_x, n_y)
237.parameters = initialize_parameters()
238.Z3 = forward_propagation(X, parameters)
239.cost = compute_cost(Z3, Y)
